### PR TITLE
Cleanup parmatch

### PR DESCRIPTION
**Pull request by gasche**
I'm creating this PR to discuss @trefis' efforts to understand and clarify the parmatch code.

### CODE REVIEW

**Review by maranget — COMMENTED** [review]
Coucou

**Comment by trefis** [comment]
Hum @gasche , I rebased my branch and pushed...
If you update trunk on this repo to rev c427e583862fdf2812e9730a7098c68887c100cc the commit list should become reasonable again.
Sorry.

**Comment by gasche** [comment]
I had to change base twice to get github to update the PR as expected (looks like they only monitor the pull-from branch...), but it worked. On the other hand, I'm in holidays right now, so don't expect fast reactions :-)

**Comment by trefis** [comment]
As a consequence of the rebase, all the comments are "lost" (i.e. they are on commits not referenced on this PR anymore).
I reproduced them on the relevant new commits.

Sorry for the email spam this must have generated.
I'll refrain from rebasing this branch from now on.

File: typing/parmatch.ml [hunk 1]
```diff
@@ -2148,26 +2149,32 @@ type amb_row = { row : pattern list ; varsets : IdSet.t list; }
...
 
-and simplify_head_pat r p ps varsets k = match p.pat_desc with
-| Tpat_alias (p,x,_) -> simplify_head_pat (IdSet.add x r) p ps varsets k
-| Tpat_var (x,_) ->
-    (omega, { row = ps; varsets = IdSet.add x r :: varsets; }) :: k
-| Tpat_or (p1,p2,_) ->
-    simplify_head_pat r p1 ps varsets (simplify_head_pat r p2 ps varsets k)
-| _ ->
-    (p, { row = ps; varsets = r :: varsets; }) :: k
+and simplify_head_pat head_bound_variables p ps varsets k =
```
Comment (trefis) [inline typing/parmatch.ml]: @gasche said
> We could also use `head_variables` (of course they are bounds in pattern, I'd say) or `head_varset` if we want something shorter.

To which I answered:
> Your choice. I personally don't mind the long name.

**Comment by gasche** [comment]
> As a consequence of the rebase, all the comments are "lost" (i.e. they are on commits not referenced on this PR anymore).

I think this is my fault for commenting before I created the PR. If I comment on the PR now, github should be able to track comments across rebases.

**Comment by gasche** [comment]
This PR has been superseded by https://github.com/ocaml/ocaml/pull/1488 and https://github.com/ocaml/ocaml/pull/1560, both merged now. Thanks!

**Pull request closed without merging**